src/main/java/seedu/address/network/Network.java

```diff
@@ -1,15 +1,43 @@
 package seedu.address.network;
 
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Optional;
+import java.util.logging.Logger;
+
+import org.apache.http.client.fluent.Request;
+
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.commons.util.StringUtil;
+
 /**
  * Manager of Network component
  */
 public abstract class Network {
+    private static final Logger logger = LogsCenter.getLogger(Network.class);
 
-    public static String fetch(String targetUrl) {
-        return lookupUrl(targetUrl);
+    public static Optional<InputStream> fetchAsStream(String url) {
```

Comment:
I think we should return an `InputStream` instead of an `Optional`, and rethrow the exception upwards.

Reply:
Done.